File: daily/cpp/201_bitwise_and_numbers_range.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
 public:
  int rangeBitwiseAnd(int left, int right) {
    int rightShift = 0;
    while (left != right) {
      left >>= 1;
      right >>= 1;
      rightShift++;
    }

    return (left << rightShift);
  }
};

File: daily/rust/201_bitwise_and_numbers_range.rs
impl Solution {
    pub fn range_bitwise_and(left: i32, right: i32) -> i32 {
        let mut right_shift = 0;
        let (mut leftt, mut rightt) = (left, right);
        while leftt != rightt {
            leftt = leftt >> 1;
            rightt = rightt >> 1;
            right_shift += 1;
        }

        leftt << right_shift
    }
}
